cheatsheets/Protect_FileUpload_Against_Malicious_File.md

```diff
@@ -1,597 +1,138 @@
+- [Java Code Snippets](#java-code-snippets)
 
-- Microsoft Office document: Word/Excel/PowerPoint using [VBA Macro](https://en.wikipedia.org/wiki/Visual_Basic_for_Applications) and [OLE package](https://en.wikipedia.org/wiki/Object_Linking_and_Embedding).
-- Adobe PDF document: Insert malicious code as attachment.
-- Images: Malicious code embedded into the file or use of binary file with image file extension.
+## File Upload Threats
 
-# Approaches
+In order to assess and know exactly what controls to implement, knowing what you're facing is essential to protect your assets. The following sections will hopefully showcase the risks accompanying the file upload functionality.
 
-Based on this context, the goals here are:
+### Malware Upload
 
-- For Word/Excel/PowerPoint/Pdf documents: Detect when a document contains "code"/OLE package, if it's the case then block the upload process.
-- For Images document: Sanitize incoming image using re-writing approach and then disable/remove any "code" present (this approach also handle case in which the file sent is not an image).
+The attacker delivers a file for malicious intent, such as:
 
-Remarks:
+1. Call the file to execute code (_e.g._ [ImageTrick Exploit](https://imagetragick.com/))
+2. Use the file for phishing (_e.g._ careers form)
+3. Send ZIP bombs, or simply huge files in a way to fill the server storage which hinders the server's availability
+4. Overwrite an existing file on the system
+5. Client-side active content that could lead to all types of client-side attacks if the file is publicly retrievable
 
-- It's technically possible to perform sanitizing on Word/Excel/PowerPoint/PDF documents but we have chosen here the option to block them in order to avoid the risk of missing any evasion techniques and then let pass one evil document. The following [site](https://www.greyhathacker.net/?p=872) show how many way exists to embed Macro into a Microsoft Office documents.
-- The other reason why we have chosen the blocking way is that for Word/Excel/PowerPoint, changing document format (for example by saving any document to `DOCX/XSLX/PPTX/PPSX` formats in order to be sure that no Macro can be executed) can have impacts or cause issues on document structure/rendering depending on the API used.
+### Public File Retrieval
 
-# Cases
+If the file uploaded is publicly retrievable, additional threats can be addressed:
 
-## Common codes
+1. Public disclosure of other files
+2. Initiate a DoS attack by requesting lots of files. Requests are small, yet responses are much larger
 
-The following codes are shared by the code snippets proposed into the rest of this article.
+## File Upload Protection
 
-Interfaces:
+### Extension Validation
 
-*DocumentDetector*
+Ensure that the validation occurs after decoding the file name, and that a proper filter is set in place (_e.g._ `\.png` vs `\.png\z`; etc.). Refer to the [Input Validation CS](Input_Validation_Cheat_Sheet.md) for more information.
```

Comment:
You often get bugs around this, so giving a few specific warnings here might be a good idea about:

* Double extensions (.jpg.php)
* Null bytes (.php%00.jpg)
* Bad regex (such as a regex for `.html$`, which could match executable formats like `.cshtml`).

Reply:
Can definitely mention them, just thought of keeping it small.